### PR TITLE
add database-global compact operation

### PR DESCRIPTION
**Pull request by jsteemann**

### CODE REVIEW

File: 3.5/data-modeling-databases-working-with.md [hunk 1]
```diff
@@ -220,6 +220,29 @@ database. The *_system* database itself cannot be dropped.
...
+The optional *options* attribute can be used to get more control over the 
+compaction. The following attributes can be used in it: 
+
+* *changeLevel*: whether or not compacted data should be moved to the minimum
+  possible level. The default value is *false*.
+* *compactBottomMostLevel*: whether or not to compact the bottommost level of
+  data. The default value is *false*.
+
+**Warning**: this command can cause a full rewrite of all data in all databases, 
+which may take long for large databases. It should thus only be used with care.
```
Comment (jsteemann) [inline 3.5/data-modeling-databases-working-with.md:241]:
```suggestion
**Warning**: this command can cause a full rewrite of all data in all databases, 
which may take very long for large databases. It should thus only be used with care.
and only when additional I/O load can be tolerated for a prolonged time.
```

File: 3.6/data-modeling-databases-working-with.md [hunk 2]
```diff
@@ -227,6 +227,29 @@ database. The *_system* database itself cannot be dropped.
...
+The optional *options* attribute can be used to get more control over the 
+compaction. The following attributes can be used in it: 
+
+* *changeLevel*: whether or not compacted data should be moved to the minimum
+  possible level. The default value is *false*.
+* *compactBottomMostLevel*: whether or not to compact the bottommost level of
+  data. The default value is *false*.
+
+**Warning**: this command can cause a full rewrite of all data in all databases, 
+which may take long for large databases. It should thus only be used with care.
```
Comment (jsteemann) [inline 3.6/data-modeling-databases-working-with.md:248]:
```suggestion
**Warning**: this command can cause a full rewrite of all data in all databases, 
which may take long for large databases. It should thus only be used with care
and only when additional I/O load can be tolerated for a prolonged time.
```

File: 3.7/data-modeling-databases-working-with.md [hunk 3]
```diff
@@ -227,6 +227,29 @@ database. The *_system* database itself cannot be dropped.
...
+The optional *options* attribute can be used to get more control over the 
+compaction. The following attributes can be used in it: 
+
+* *changeLevel*: whether or not compacted data should be moved to the minimum
+  possible level. The default value is *false*.
+* *compactBottomMostLevel*: whether or not to compact the bottommost level of
+  data. The default value is *false*.
+
+**Warning**: this command can cause a full rewrite of all data in all databases, 
+which may take long for large databases. It should thus only be used with care.
```
Comment (jsteemann) [inline 3.7/data-modeling-databases-working-with.md:248]:
```suggestion
**Warning**: this command can cause a full rewrite of all data in all databases, 
which may take long for large databases. It should thus only be used with care
and only when additional I/O load can be tolerated for a prolonged time.
```

File: 3.8/data-modeling-databases-working-with.md [hunk 4]
```diff
@@ -227,6 +227,29 @@ database. The *_system* database itself cannot be dropped.
...
+The optional *options* attribute can be used to get more control over the 
+compaction. The following attributes can be used in it: 
+
+* *changeLevel*: whether or not compacted data should be moved to the minimum
+  possible level. The default value is *false*.
+* *compactBottomMostLevel*: whether or not to compact the bottommost level of
+  data. The default value is *false*.
+
+**Warning**: this command can cause a full rewrite of all data in all databases, 
+which may take long for large databases. It should thus only be used with care.
```
Comment (jsteemann) [inline 3.8/data-modeling-databases-working-with.md:248]:
```suggestion
**Warning**: this command can cause a full rewrite of all data in all databases, 
which may take long for large databases. It should thus only be used with care
and only when additional I/O load can be tolerated for a prolonged time.
```

File: 3.8/data-modeling-databases-working-with.md [hunk 5]
```diff
@@ -227,6 +227,30 @@ database. The *_system* database itself cannot be dropped.
...
+  possible level. The default value is *false*.
+* *compactBottomMostLevel*: whether or not to compact the bottommost level of
+  data. The default value is *false*.
+
+**Warning**: this command can cause a full rewrite of all data in all databases, 
+which may take long for large databases. It should thus only be used with care
+and only when additional I/O load can be tolerated for a prolonged time.
+
+This command requires superuser access and is only available for the RocksDB
+storage engine.
```
Comment (jsteemann) [inline 3.8/data-modeling-databases-working-with.md:252]:
```suggestion
This command requires superuser access.
```

File: 3.7/data-modeling-databases-working-with.md [hunk 6]
```diff
@@ -227,6 +227,30 @@ database. The *_system* database itself cannot be dropped.
...
+  possible level. The default value is *false*.
+* *compactBottomMostLevel*: whether or not to compact the bottommost level of
+  data. The default value is *false*.
+
+**Warning**: this command can cause a full rewrite of all data in all databases, 
+which may take long for large databases. It should thus only be used with care
+and only when additional I/O load can be tolerated for a prolonged time.
+
+This command requires superuser access and is only available for the RocksDB
+storage engine.
```
Comment (jsteemann) [inline 3.7/data-modeling-databases-working-with.md:252]:
```suggestion
This command requires superuser access.
```

**Review by Simran-B — APPROVED** [review]

**Comment by Simran-B** [comment]
REST API documentation: https://github.com/arangodb/arangodb/pull/12666

**Pull request merged**